packages/flutter/test/widgets/sliver_cross_axis_group_test.dart

```diff
@@ -806,6 +808,65 @@ void main() {
     // If renderHeader._lastStartedScrollDirection is not ScrollDirection.forward, then we shouldn't see the header at all.
     expect((renderHeader.parentData! as SliverPhysicalParentData).paintOffset.dy, equals(0.0));
   });
+
```

Comment:
Both the cross axis group and the main axis group tests here pass without the change to the paint method in the group widgets. They don't validate the change.

What they are checking is that the SliverToBoxAdapters are painting/not painting as expected, but not the sliver groups.

Reply:
How would I solve this then? I'd need to verify that SliverCrossAxisGroup and SliverMainAxisGroup draws exactly two boxes, but I'm not sure how to do this.

Reply:
There are a couple different ways to do this. You can look up the Sliver*AxisGroup in the same way you are looking up the SliverToBoxAdapter.
You could mock out the class and increment a counter every time it paints.
There are probably a lot of different examples in existing tests if that helps. :)